### ISSUE TITLE
Unable to use pipeline aggregation

### ISSUE THREAD

**Issue by Tyagi-Gaurav** [sksamuel/elastic4s#838]
I am trying to use pipeline aggregation for sum as in the example shown below.

```
"aggs" : {
        "sales_per_month" : {
            "date_histogram" : {
                "field" : "date",
                "interval" : "month"
            },
            "aggs": {
                "sales": {
                    "sum": {
                        "field": "price"
                    }
                }
            }
        },
        "max_monthly_sales": {
            "max_bucket": {
                "buckets_path": "sales_per_month>sales" 
            }
        }
    }
```

The current API supports adding only `AggregationDefinition` at the top level, while the pipeline aggregation needs to be added as a sub aggregation. `PipelineAggregationDefinition` and `AggregationDefinition` have different class hierarchies. In the previous versions, the aggregation definition was using an `AbstractAggregationDefinition` but its support seems to have been dropped. 

Is there any example on how to construct the above aggregation using elastic4s dsl ?

**Comment by sksamuel** [comment]
Are you using the HTTP client or TCP ?

**Comment by Tyagi-Gaurav** [comment]
I am using TCPClient.

**Comment by sksamuel** [comment]
Fix will be in 5.3.1 and I'll backport to 5.2.x
